File: Osnovne skole/2024/Dronovi/Dronovi.py
import math
sqrt_2 = math.sqrt(2)

def parse(dir):
    if dir == "S":
        return 0.0, 1.0
    elif dir == "J":
        return 0.0, -1.0
    elif dir == "I":
        return 1.0, 0.0
    elif dir == "Z":
        return -1.0, 0.0
    elif dir == "SI":
        return 1.0/sqrt_2, 1.0/sqrt_2
    elif dir == "SZ":
        return -1.0/sqrt_2, 1.0/sqrt_2
    elif dir == "JI":
        return 1.0/sqrt_2, -1.0/sqrt_2
    elif dir == "JZ":
        return -1.0/sqrt_2, -1.0/sqrt_2
    
xm = 0.0
ym = 0.0
xh = 1.0
yh = 0.0

M = int(input())
for i in range(M):
    dir, value = input().split()
    value = float(value)

    x_sign, y_sign = parse(dir)
    xm += x_sign * value
    ym += y_sign * value

H = int(input())
for i in range(H):
    dir, value = input().split()
    value = float(value)

    x_sign, y_sign = parse(dir)
    xh += x_sign * value
    yh += y_sign * value

diff = math.sqrt((xm - xh) ** 2 + (ym - yh) ** 2)
print(diff)

File: Osnovne skole/2024/Dronovi/Dronovi.cpp
#include<iostream>
#include<math.h>
using namespace std;

void parse(string dir, double &x_sign, double &y_sign) {
    if(dir == "S") { x_sign = 0; y_sign = 1; }
    else if(dir == "J") { x_sign = 0; y_sign = -1; }
    else if(dir == "I") { x_sign = 1; y_sign = 0;  }
    else if(dir == "Z") { x_sign = -1; y_sign = 0; }
    else if(dir == "SI") { x_sign = 1./sqrt(2); y_sign = 1./sqrt(2); }
    else if(dir == "SZ") { x_sign = -1./sqrt(2); y_sign = 1./sqrt(2); }
    else if(dir == "JI") { x_sign = 1./sqrt(2); y_sign = -1./sqrt(2); }
    else if(dir == "JZ") { x_sign = -1./sqrt(2); y_sign = -1./sqrt(2); }
}

int main() {
    int M, H;
    double xm = 0, ym = 0;
    double xh = 1, yh = 0;

    cin >> M;
    for(int i = 0 ; i < M; i++) {
        string dir; float value;
        cin >> dir >> value;
        double x_sign, y_sign;
        parse(dir, x_sign, y_sign);
        xm += x_sign * value; 
        ym += y_sign * value;
    }

    cin >> H;
    for(int i = 0 ; i < H; i++) {
        string dir; float value; 
        cin >> dir >> value;
        double x_sign, y_sign;
        parse(dir, x_sign, y_sign);
        xh += x_sign * value;
        yh += y_sign * value;
    }

    double diff = sqrt((xm - xh) * (xm - xh) + (ym - yh) * (ym - yh));
    cout << diff << endl;
    return 0;
}